src/components/header/header_links/header_link.stories.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Meta, StoryObj } from '@storybook/react';
+
+import { disableStorybookControls } from '../../../../.storybook/utils';
+import { EuiHeaderLink, EuiHeaderLinkProps } from './header_link';
+
+const meta: Meta<EuiHeaderLinkProps> = {
+  title: 'EuiHeaderLink',
+  component: EuiHeaderLink,
+  argTypes: {
+    flush: {
+      options: [undefined, 'left', 'right', 'both'],
+    },
+    iconType: { control: 'text' },
+    target: { control: 'text' },
+  },
+  // Component defaults
+  args: {
+    type: 'button',
+    size: 'm',
+    iconSize: 'm',
+    iconSide: 'left',
+    isDisabled: false,
+    isLoading: false,
+    isSelected: false,
+    isActive: false,
+  },
+};
```

Comment:
I'm honestly a little confused as to why we even export this component other than syntactical sugar, since it's basically just an `EuiButtonEmpty` 😅 

Since `EuiHeaderLinkProps` comes with a ton of optional props that `EuiHeaderLink` doesn't _necessarily_ need, I think we could simplify the playground for this to just the prop that _is_ unique to the component, i.e.

```suggestion
const meta: Meta<{ isActive?: boolean }> = {
  title: 'EuiHeaderLink',
  component: EuiHeaderLink,
  // Component defaults
  args: {
    isActive: false,
  },
};
```

Reply:
@cee-chen  Yeah, I was debating if those wrappers should only define their specific props or all available. 😅 
My decision was based on the following thinking: The props will still be available in the playground and the controls table (just not set with defaults) as they are defined as props for the wrapper to pass down to the inner component. And the inner component does have those defaults applied so I thought it might make sense to show what is set by default when using the wrapper component.

But I totally understand keeping it scoped to the actual specific props! 
I'm wondering if the playground controls are confusing if all props are available but unset? It's actually possible to group controls by category - maybe separating component specific and additional props could be an idea? 🤔 

![Screenshot 2024-03-07 at 10 45 05](https://github.com/elastic/eui/assets/44670957/ae9d0d84-4678-4f26-bd30-873710d8b5b8)

Reply:
> I'm wondering if the playground controls are confusing if all props are available but unset? It's actually possible to group controls by category - maybe separating component specific and additional props could be an idea? 🤔

Ooo I didn't even realize this was an option! Yes, I'd be fine with that if so. Basically I just want some way of differentiating between "these are the props most relevant/most used by this component, and these are props that are inherited and available for use (but significantly less common)".

Reply:
The caveat with this is that it's not automated, we have to define it manually for the props in question to be added to another category via [`table.category`](https://storybook.js.org/docs/api/arg-types#tablecategory), but I guess we could add a util similar to `hideStorybookControls` for it?

We could introduce it in a separate PR and update existing stories that could benefit from using it?

Reply:
Ah darnit, Storybook's controls APIs feel so cumbersome. Yep, that makes total sense, let's save it for another PR!